scripts/generate_errors.pl

```diff
@@ -85,7 +89,7 @@
         $description =~ s/^\s+//;
         $description =~ s/\n( *\*)? */ /g;
         $description =~ s/\.?\s+$//;
-        push @matches, [$name, $value, $description];
+        push @matches, [$name, $value, $description, grep(/^.*private\/[^\/]+$/, $file)];
```

Comment:
This technically works because Perl is weird, but it's fragile. `grep(/RE/, $scalar)` is either the list `($scalar)` if the regexp matches or the empty list otherwise. If it matches, we end up with a 4-element list whose last element is `$scalar`, which is truthy because `$scalar` will never be something like `"0 but private/true.h"`. If it doesn't match, then we end up with a 3-element list, and below we unpack 4 elements so `$is_private_header` becomes `undef` which is falsy.

Please rewrite this to always construct a 4-element list. For example:
```suggestion
        push @matches, [$name, $value, $description, scalar($file =~ /^.*private\/[^\/]+$/)];
```
(`scalar grep …` would also work, because that counts the number of matching arguments, but I think using the match operator, which counts matches within the one string, is clearer.)